src/main/java/Duke.java

```diff
@@ -1,10 +1,202 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 public class Duke {
+    private static ArrayList<Task> currList = new ArrayList<>();
+    private static final String FILE_PATH = "./data/duke.txt";
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetUser();
+
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            try {
+                String currInput = sc.nextLine().trim();
+                processInput(currInput);
+                saveTasksToFile();
+            } catch (DukeException e) {
+                System.out.println(">  " + e.getMessage());
+            } catch (NumberFormatException e) {
+                System.out.println(">  Please provide a valid task number.");
+            }
+            System.out.println("------------------------------------------------");
+        }
+    }
```

Comment:
Your `main` function is concise and easy to understand! You can consider putting the error message for `NumberFormatException` in the exception object instead of hardcoding the message.